### ISSUE TITLE
Feature Request: Double Clicking the Top Left part of the vscode window does not close it, unlike other windows native applications. Can this be implemented?

### ISSUE THREAD

**Issue by animehunter123** [microsoft/vscode#159010]
Hello,

Normally when you double click the top left of a Window, the application closes. 

When you double click vscode, the application maximizes.

Can VSCode have this capability as well to close?

![output](https://user-images.githubusercontent.com/42163211/186343544-4f43baae-9965-4c08-b91a-e3397ff9d743.gif)

**Comment by bpasero** [comment]
We have logic to detect and handle double click in the custom title to restore / maximize the window but we missed to handle this case where on Windows it indeed seems to close the app.

For this to work we probably have to detect a click on the icon and then call a different method to close instead of the current behaviour.

Code pointers:

https://github.com/microsoft/vscode/blob/b65e8ba81b7a3b56e59b02da6021c2abead6659f/src/vs/platform/native/common/native.ts#L70

//cc @sbatten

**Comment by bpasero** [comment]
Oh interesting, that code only runs on macOS, then I wonder what triggers this 🤔

**Comment by animehunter123** [comment]
Yes, very interesting. 

For other Windows apps, the menu appears when you single left click the top window as well. In VSCode when you Single Click the top left, nothing happens (In windows 10, all versions, latest VSCode)

**Comment by bpasero** [comment]
Yeah that is expected, we draw the title custom entirely. We need to implement those native integrations on top. Maybe you report as separate issue.

**Comment by sbatten** [comment]
I think you want this setting `window.doubleClickIconToClose`
